src/stores/useUsdcStore.ts

```diff
@@ -0,0 +1,51 @@
+import { Web3Provider } from '@ethersproject/providers';
+import { Web3ReactContextInterface } from '@web3-react/core/dist/types';
+import BigNumber from 'bignumber.js';
+import ERC20Abi from 'config/abi/IERC20.json';
+import { banks } from 'config/constants/banks';
+import { Address } from 'config/constants/types';
+import { getContract } from 'utils/contractHelper';
+import { isLocalhost } from 'utils/misc';
+import createStore from 'zustand';
+import { combine } from 'zustand/middleware';
+
+const initialState = {
+  usdcBalance: 0,
+  isLoading: true,
+};
+
+export const useUsdcStore = createStore(
+  combine(initialState, (set, _get) => ({
+    initialize: async ({
+      chainId,
+      library,
+      account,
+    }: Required<
+      Pick<Web3ReactContextInterface<Web3Provider>, 'chainId' | 'library' | 'account'>
+    >) => {
+      try {
```

Comment:
It was a pain in the ass to figure out how to do this, the main / beta branch is a React hook soup, but dig deep enough and you see the Web3 libraries are pretty straight forward. The only part of this code that is still ugly is the way we get the bank and underlying addresses. Much over-engineering to overcome in this project...